tests/functional/firstrun_sign_in.js

```diff
@@ -66,6 +67,31 @@ define([
         .end();
     },
 
+    'sign in with an existing account with the `haltAfterSignIn=true` query parameter': function () {
+      var self = this;
+
+      return FunctionalHelpers.openPage(this, NO_REDIRECT_URL, '#fxa-signin-header')
+        .execute(listenForFxaCommands)
+
+        .then(respondToWebChannelMessage(self, 'fxaccounts:can_link_account', { ok: true } ))
+
+
+        .then(function () {
+          return FunctionalHelpers.fillOutSignIn(self, email, PASSWORD);
+        })
+
+        .then(FunctionalHelpers.testIsBrowserNotified(self, 'fxaccounts:can_link_account'))
+        .then(FunctionalHelpers.testIsBrowserNotified(self, 'fxaccounts:login'))
+
+        // Some settings do not work in an iframe due to x-frame and
+        // same-origin policies. Allow the firstrun flow to decide whether
+        // they want to display the settings page after the `login` message
+        // is sent. If `haltAfterSignIn` is set to true, the firstrun page
+        // will take care of displaying an update to the user.
```

Comment:
This comment seems very elaborate and beyond the scope of the test; since this reasoning is already outlined in the code itself, perhaps the corresponding test comment can be a shorter summary like "don't go to settings page if haltAfterSignIn=true".